buildSrc/src/main/java/org/elasticsearch/gradle/JdkDownloadPlugin.java

```diff
@@ -57,218 +38,101 @@
 
     @Override
     public void apply(Project project) {
-        NamedDomainObjectContainer<Jdk> jdksContainer = project.container(
-            Jdk.class,
-            name -> new Jdk(name, project.getConfigurations().create("jdk_" + name), project.getObjects())
-        );
-        project.getExtensions().add(EXTENSION_NAME, jdksContainer);
-
-        project.afterEvaluate(p -> {
-            for (Jdk jdk : jdksContainer) {
-                jdk.finalizeValues();
+        Attribute<String> artifactType = Attribute.of("artifactType", String.class);
+        project.getDependencies().getArtifactTypes().maybeCreate(ArtifactTypeDefinition.ZIP_TYPE);
 
-                // depend on the jdk directory "artifact" from the root project
-                DependencyHandler dependencies = project.getDependencies();
-                Map<String, Object> depConfig = new HashMap<>();
-                depConfig.put("path", ":"); // root project
-                depConfig.put(
-                    "configuration",
-                    configName("extracted_jdk", jdk.getVendor(), jdk.getVersion(), jdk.getPlatform(), jdk.getArchitecture())
-                );
-                project.getDependencies().add(jdk.getConfigurationName(), dependencies.project(depConfig));
+        project.getDependencies().registerTransform(UnzipTransform.class, noneTransformSpec -> {
+            noneTransformSpec.getFrom().attribute(artifactType, ArtifactTypeDefinition.ZIP_TYPE);
+            noneTransformSpec.getTo().attribute(artifactType, ArtifactTypeDefinition.DIRECTORY_TYPE);
+        });
 
-                // ensure a root level jdk download task exists
-                setupRootJdkDownload(project.getRootProject(), jdk);
-            }
+        ArtifactTypeDefinition tarArtifactTypeDefinition = project.getDependencies().getArtifactTypes().maybeCreate("tar.gz");
```

Comment:
Should we register `.tgz` as well?

Reply:
Seems for this particular jdk download its not required.